web/server/codechecker_server/api/mass_store_run.py

```diff
@@ -529,6 +575,66 @@ def __add_file_content(
                 # the meantime.
                 session.rollback()
 
+    def __store_checker_identifiers(self, checkers: Set[Tuple[str, str]]):
+        """
+        Stores the identifiers "(analyzer, checker_name)" in the database into
+        a look-up table where each unique checker is given a unique numeric
+        identifier.
+
+        Due to the use of an M-to-N connection table
+        (see `AnalysisInfoChecker`) one side of the joins must have their IDs
+        eagerly populated, otherwise the Python bindings will fail.
+        However, eager population will result in exceptions that a flush was
+        created before the transaction was complete.
+
+        Moreover, this is performed separately from the storing of the details
+        of a run to reduce contention if two parallel stores, especially across
+        server instances (in a distributed/load-balanced environment) want to
+        store the same identifier(s).
+        """
+        max_tries, tries, wait_time = 3, 0, timedelta(seconds=30)
```

Comment:
`(analyzer_name, checker_name)` has a `UNIQUE` constraint on them. Every conceptually existing checker will need to have at most **one** `checker_id` for the look-up to work. If two `store`s happen concurrently and they store the same set of checker names (highly likely), the storage that ends up `COMMIT`ting later will explode due to the constraint violation. This is an existing issue with everything that is supposedly stored as a "product-global" information, e.g., [files and contents](https://github.com/Ericsson/codechecker/blob/v6.23.0/web/server/codechecker_server/api/mass_store_run.py#L132), [review statuses](https://github.com/Ericsson/codechecker/blob/v6.23.0/web/server/codechecker_server/api/mass_store_run.py#L1238-L1259).

In fact, in this patch, I removed the above second logic, due to having previously moved review statuses into individual `report` rows. But the general issue with the logic now comes up again with the introduction of a new product-global information.

```py3
# TODO: Since review status is stored in "reports" table and
# "review_statuses" table is not written during storage, this
# multiple trials should be unnecessary.
```

Reply:
The only downside of, essentially pre-committing checker names, is that **if** a store introduces a new checker name, let's say `foo/bar`, and the report storage fails, only the report storage is rolled back, because the ID for the checkers are already there and committed. So, we will store a little bit of _potentially_ unused data in the database. But it's two generally small strings with a number, so like... guesstimateable to be at most 256 bytes-ish.